Adafruit_FXOS8700.cpp

```diff
@@ -321,17 +324,21 @@ void Adafruit_FXOS8700::getSensor(sensor_t *accelSensor, sensor_t *magSensor) {
     non-standard .getEvent call with two parameters should
     generally be used with this sensor.
 
-    @param    accelEvent
+    @param    singleSensorEvent
               A reference to the sensors_event_t instances where the
-              accelerometer data should be written.
+              accelerometer or magnetometer data should be written.
 
     @return True if the event read was successful, otherwise false.
 */
 /**************************************************************************/
-bool Adafruit_FXOS8700::getEvent(sensors_event_t *accelEvent) {
-  sensors_event_t accel;
-
-  return getEvent(accelEvent, &accel);
+bool Adafruit_FXOS8700::getEvent(sensors_event_t *singleSensorEvent) {
+  sensors_event_t dummy;
+  switch (_mode) {
+  case ACCEL_ONLY_MODE:
+    return getEvent(singleSensorEvent, &dummy);
+  case MAG_ONLY_MODE:
+    return getEvent(&dummy, singleSensorEvent);
+  }
```

Comment:
You'll need to provide a default return to make the compiler happy.

Reply:
Added default case returning false.